xalia/Win32/HwndTrackBarProvider.cs

```diff
@@ -0,0 +1,131 @@
+            "reversed",
+            "tooltips",
+            "nothumb",
+            "fixedlength",
+            "enableselrange",
+            "noticks",
+            "both"
+        };
+
+        static Dictionary<string,int> style_flags;
+
+        static HwndTrackBarProvider()
+        {
+            style_flags = new Dictionary<string, int>();
+            for (int i=0; i<style_names.Length; i++)
+            {
+                if (style_names[i] is null)
+                    continue;
+                style_flags[style_names[i]] = 0x1000 >> i;
+            }
+        }
+
+        public override UiDomValue EvaluateIdentifier(UiDomElement element, string identifier, HashSet<(UiDomElement, GudlExpression)> depends_on)
+        {
+            switch (identifier)
+            {
+                case "is_hwnd_track_bar":
+                case "is_hwnd_trackbar":
+                    return UiDomBoolean.True;
+            }
+            return UiDomUndefined.Instance;
+        }
+
+        public override UiDomValue EvaluateIdentifierLate(UiDomElement element, string identifier, HashSet<(UiDomElement, GudlExpression)> depends_on)
+        {
+            switch (identifier)
+            {
+                case "slider":
+                    return UiDomBoolean.True;
+                case "role":
+                case "control_type":
+                    return role;
+                case "autoticks":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & TBS_AUTOTICKS) != 0);
+                case "vertical":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & TBS_VERT) != 0);
+                case "horizontal":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & TBS_VERT) == 0);
+                case "left":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & (TBS_VERT|TBS_LEFT|TBS_BOTH)) == (TBS_VERT|TBS_LEFT));
+                case "right":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & (TBS_VERT|TBS_LEFT|TBS_BOTH)) == TBS_VERT);
+                case "top":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & (TBS_VERT|TBS_TOP|TBS_BOTH)) == TBS_TOP);
+                case "bottom":
+                    depends_on.Add((element, new IdentifierExpression("win32_style")));
+                    return UiDomBoolean.FromBool((HwndProvider.Style & (TBS_VERT|TBS_TOP|TBS_BOTH)) == 0);

```

Comment:
It probably depends on what the properties are used for. At the moment there is no use for them, and they're only there because I wanted to include all the styles.

I can't think of any likely use case.

Reply:
I guess if these properties are meant to expose if the ticks are exclusively on the left/right/top/bottom of the scrollbar this is fine. Still, if we're exposing those, it seems like we should have some equivalent in `TBS_BOTH`.

If nothing is using this it's not a big deal, just thought it was worth pointing out.

Reply:
"both" is in here, it's easier because it's a static flag in a contiguous range of static flags: https://github.com/madewokherd/xalia/pull/13/files#diff-c6159a124159f456728f7070047655e086dda97dfd11f202b5bbdbb891610703R118

Reply:
Oh doh, my bad. Never mind then. :D